sympy/geometry/polygon.py

```diff
@@ -402,6 +403,59 @@ def centroid(self):
             cy += v*(y1 + y2)
         return Point(simplify(A*cx), simplify(A*cy))
 
+    def second_moment_of_area(self, p=None):
+        """Second and product moment of area of polygon.
+
+        Parameters
+        ==========
+
+        p : Point, None
+            p is the point about which second moment of area is to be find. Default value is None
+
+        Notes
+        =====
+
+        function will return second moment of area about the axis passing through centroid
+        if point is not given othervise about the axis passing through the p point.
+
+        Examples
+        ========
+
+        >>> from sympy import Point,Polygon,symbols
+        >>> a, b = symbols('a, b')
+        >>> p1, p2, p3, p4, p5 = [(0, 0), (a, 0), (a, b), (0, b), (a/3, b/3)]
```

Comment:
I thought that SymPy has an actual Point2D object. These are just tuples of specifiable objects. This should be clear in the doc string. Why doesn't it work with Point2D?

Reply:
it is working with both Point2D object as well.